app/src/main/java/com/github/sdp/ratemyepfl/database/ReviewRepositoryImpl.kt

```diff
@@ -103,43 +103,31 @@ class ReviewRepositoryImpl(val repository: RepositoryImpl<Review>) : ReviewRepos
             .mapNotNull { obj -> obj.toReview()?.withId(obj.id) }
     }
 
-    override suspend fun addUidInArray(fieldName: String, id: String, uid: String) {
-        when (fieldName) {
-            LIKERS_FIELD_NAME -> addUpVote(id, uid).await()
-            DISLIKERS_FIELD_NAME -> addDownVote(id, uid).await()
-            else -> throw DatabaseException("$fieldName is not an Array")
-        }
-    }
-
-    override suspend fun removeUidInArray(fieldName: String, id: String, uid: String) {
-        when (fieldName) {
-            LIKERS_FIELD_NAME -> removeUpVote(id, uid).await()
-            DISLIKERS_FIELD_NAME -> removeDownVote(id, uid).await()
-            else -> throw DatabaseException("$fieldName is not an Array")
-        }
-    }
-
-    override fun addUpVote(reviewId: String, userId: String) = repository
-        .update(reviewId) { review ->
+    override suspend fun addUpVote(reviewId: String, userId: String) {
+        repository.update(reviewId) { review ->
             if (!review.likers.contains(userId))
                 review.copy(likers = review.likers.plus(userId))
             else review
-        }
+        }.await()
+    }

```

Comment:
Instead of waiting you could call the updateList function in the view model as a callback. This holds also for downVote and karma, but both solution are good.

Reply:
In the end, all the solutions do the same thing I think. I personally like Julien's code as all the functions are suspend, we can call all of them in just one _viewModelScope.launch_